client/src/store/features/users/usersSlice.js

```diff
@@ -205,4 +205,24 @@ export function updateUserProfile(userId, userToken) {
   }
 }
 
+export const signupUser = createAsyncThunk(
+  'users/signupUser',
+  async (dispatch, { username, email, password }, thunkAPI) => {
+    try {
+      const response = await userService.signUp(username, email, password)
+      const data = await response.json()
```

Comment:
We are using `axios`, so there is no need to use `respnse.json. Just return `response.data`.